contracts/access/AccessControl.sol

```diff
@@ -164,12 +164,17 @@ abstract contract AccessControl is Context {
      * event. Note that unlike {grantRole}, this function doesn't perform any
      * checks on the calling account.
      *
-     * Requirements:
-     *
-     * - this function can only be called from a constructor.
+     * [WARNING]
+     * ====
+     * This function should only be called from the constructor when setting
+     * up the initial roles for the system.
+     *
+     * It being callable by other means goes against the spirit of
+     * {AccessControl}, as there would be ways to grant roles not tracked by
+     * {getRoleMemberCount} and {getRoleMember}.
```

Comment:
I don't understand this. Can you explain?

Reply:
Say you have the following code:

```solidity
function giveRole(to, role) {
   if (condition) {
      _setupRole(role, to);
   }
}
```

Then there is a way to give roles without any account being its role admin. Part of the idea behind AccessControl is being able to determine who can grant new roles: calling this function outside of a constructor undermines this.

Reply:
What I don't understand is what `getRoleMemberCount` and `getRoleMember` have to do with that.

Reply:
Oh, those are the methods that are used to query which accounts have a role, so it's their guarantees that would be broken in a wy.

Reply:
For example, `getRoleMemberCount` returning zero for a given role would normally mean that any role for which that one is admin can no longer be granted, but usage of `_setupRole` changes that.

Reply:
Okay. I don't think that point is clear. I'll make a suggestion.